### PR TITLE
Expose applied filters through getters

### PR DESCRIPTION
**Pull request by thomasgillet**
From issue #92.

### CODE REVIEW

**Comment by esjewett** [comment]
Looks OK to me. @gordonwoodhull - any comments, given that you had some opinions on #92 ?

**Comment by esjewett** [comment]
Or rather, @gordonwoodhull had some comments on #44, which #92 is a reference too...

File: src/crossfilter.js [hunk 1]
```diff
@@ -43,6 +43,8 @@ function crossfilter() {
 
   var data = [], // the records
       n = 0, // the number of records; data.length
+      filterValue, // the value used for filtering (value, array, function or undefined)
+      filterValuePresent, // true if filterValue contains something
```
Comment (gordonwoodhull) [inline src/crossfilter.js]: I think these need to be declared at the dimension level. Each dimension has its own filter, and if I'm not mistaken, declaring them here will result in the values getting overwritten if more than one dimension has an active filter. (As well as unfiltered dimensions returning the filters of filtered dimensions.)

**Comment by gordonwoodhull** [comment]
The interface looks great, and it will be really helpful to have this property exposed.

I've noted a possible bug in the implementation. It's probably worthwhile to add a test or two that set the filter on multiple dimensions and make sure they don't interfere with each other.

File: src/crossfilter.js (hunk 1 above)
Comment (thomasgillet) [reply src/crossfilter.js]: Ooops sorry, my bad. Fixed it, with a new test as you suggested.

**Comment by thomasgillet** [comment]
I think I'll stop here for this PR, and open a new one for the `crossfilter.dimensions` method, along with other navigation methods (`dimension.groups`, access to parent...).

**Comment by esjewett** [comment]
Merged - going to be part of the 1.4.4 release.

**Comment by gordonwoodhull** [comment]
Thanks @thomasgillet!

**Comment by esjewett** [comment]
This was released in https://github.com/crossfilter/crossfilter/releases/tag/1.4.5

**Comment by Tahirhan** [comment]
Thanks for release, but in case of multi filter selection "currentFilter" method returns function. in console log response seems like below 

`ƒ (d) {
                for (var i = 0; i < filters.length; i++) {
                    var filter = filters[i];
                    if (filter.isFiltered && filter.isFiltered(d)) {
                    …`

**Comment by gordonwoodhull** [comment]
Hi @Tahirhan, that sounds like it might correct. Crossfilter does not natively support a multi selection. 

If you are using dc.js it will use `filterFunction` in order to implement the multi selection. (It uses `filterExact` if there is only one item selected.)

If you are using dc.js you should retrieve the current filters using `chart.filters()`

**Comment by Tahirhan** [comment]
Thank you @gordonwoodhull , i realized that function response is coming from dc.js . I thought that i can get filters from crossfilter object at once and i didn't know charts have filters method so thank you .

**Comment by gordonwoodhull** [comment]
I [documented these features in the wiki](https://github.com/crossfilter/crossfilter/wiki/API-Reference#dimension_currentFilter). Any and all, please check for accuracy.

**Pull request merged**